daemon/create_windows.go

```diff
@@ -2,20 +2,41 @@ package daemon
 
 import (
 	"fmt"
+	"strings"
 
 	containertypes "github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/container"
 	"github.com/docker/docker/pkg/stringid"
 	"github.com/docker/docker/volume"
 )
 
+// dedupConfigEnvVars deduplicates requested environment variables in the config
+// as they are case insensitive on Windows
+func dedupConfigEnvVars(config *containertypes.Config) {
+	deduped := make(map[string]string)
+	for _, v := range config.Env {
+		i := strings.Index(v, "=")
+		if i >= 0 {
+			name := strings.ToUpper(v[:i])
+			value := v[i+1:]
+			deduped[name] = value
+		}
+	}
+	config.Env = nil
+	for k, v := range deduped {
+		config.Env = append(config.Env, k+"="+v)
+	}
+}
```

Comment:
Hm, not sure. Basically we decided to leave the interpretation to the shell, because docker shouldn't really care how the shell handled it (especially if we don't know what the shell _is_).

Duplicating the logic of the shell could lead to a discrepancy there; what if future versions / different versions of a shell _do_ support case-sensitive env-vars, or support value "lists"?